LiteCore/Database/DatabaseImpl.cc

```diff
@@ -252,7 +252,10 @@ namespace litecore {
         // Create a new BlobStore and copy/rekey the blobs into it:
         filePath().subdirectoryNamed("Attachments_temp").delRecursive();
         auto &blobStore = getBlobStore();
+        auto previous = _config.flags;
+        _config.flags |= kC4DB_Create; // Temporarily allow creation, since we need a temporary key store
```

Comment:
This is kind of a kludge and will leave the flags incorrect if `createBlobStore` fails. 🧐 
How about adding a flags parameter to `createBlobStore` instead?

Reply:
Good point, although adding it to the signature is like saying "create it" or "REALLY create it" as the method logic.